helm-charts/templates/data-plane/gateway-components/gateway-runtime/listener-crt-secret.yaml

```diff
@@ -0,0 +1,19 @@
+{{ if or .Values.wso2.apk.dp.enabled .Values.wso2.apk.cp.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "apk-helm.resource.prefix" . }}-listener-crt-secret
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+  gw.key: LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1JSUV2Z0lCQURBTkJna3Foa2lHOXcwQkFRRUZBQVNDQktnd2dnU2tBZ0VBQW9JQkFRQ3ljMEE1dndWaVBOejYKWlBFTjFKWDAvRGFrK3VlWkx1UURpbENvTDBrU05IYUFOa21Xa3dER3FTMFVCOEtZcTRBZUlnMDdsdjVJNHBwagp6aHlkZlNNWXVuMEJOMldENUUvenJ1WGpGRGlha0xJUnRXYldJWjc5dzFPUXJGdmpFTzc2RnlCRnF4M3pBdW15CmVkckViaFBUMWlBbE9mUmxSYmpKd1VSVFdiNmtZQUF4UEdQdzRwVHBPNHVHd0ZnNFhhS1VTYTV4VlJVNVV3K0kKSkVGLzJvQ3IrK25WcmV5UlV3VEFjOUZORWlkcnRDVTB5d0pyK0FwWkRpQTBaT3NaYnl4cXF2RzQwK0pIQUk3cwpYTy9jQ0ErV1JsWkR6Yy9DbkovbjZlQytEaFd6V0xBNVd1K3FTSVVmQklXNGlyWkZpRU5WWWIzODArSm5VNlkrCllWWHpYS3FMQWdNQkFBRUNnZ0VCQUlYNHhMVVYxa0xra3JwaDRoK1NCR3NDbHp2Ukt1TDBNemI5Tm9TdHkxd2gKaXYzQ3hBaUpvTUdFN1ZUbVpwQWI0WG05OHRWVURPQkhUOFd5VGdWbWwxelBWVjMrYXZaVUNEcFlpanFyU2J1dworaDhWWTNsTGtJUFBpd05JZzB1N3dNNTlZanNydHJaazczS1F6VTgvblI4aTJNdFo4VU8rem5hWjVHUDNNb3RRClRxWWZzdlFwK3ZpM1VtWHpnejBIL3dUVFg1a1BhUkFUSzU5aFdwL050NlJBdXdDMzZrY1Y0aWN0L01tWGVxK04KK25GSTd1Sk5lWXFPQUNCS3FqdVFFdUswVjJDRXUzOHkwd3M4N2pVMlBqSHdpT29YYzFBbVZwTC9qRW5XTmNkUAp1QzVrTkQyREFlV3pSdmpsUFRxcGt4ZkowMnd0VHlKSE84ckRBN0Y1RDZFQ2dZRUExa2NxV0JFQVc5RXVkQlZwCjNjWXRzdkZKVTNRWHFFTlp5OVNNWHRpN0tmbVJJdUVnak5ZeEJrNDRtUW5QeWNHSnE2ek9tZkhaaFZHaEtFSUgKSXVBVmJGZ0IxUHRxUHYwQ2lmaXFtZUVoRzZjYVJMdTFrOC9IMWtJZ2MyUTZqcU9Ndk9PeVA3L3FpUGdBNlBNMgo5aVBtMTBIdGlMbzFGWTB0Z0hjOXNBS05DK2tDZ1lFQTFUSTVkWUhrSHhYbFF2NzNER0RLMGViN2RsMklWcDhrCjVoTTlyTTVFc0dCdm9WbS9JWElFMGJuTEVDNUNKVm9lYjFaS0xxb3hoWjg3WVVrMkhJWEN4c3FLUWYvbGd3SjkKaTJ0RE82Ym01TDJOV2JuS3pqMUVpT295SGJTNGhwczRUOHJFVjk4QjhqMzhFSUdmeEN0Z2VxWFhoeDFNNEhCRAo1ZS9GVXpzSW5sTUNnWUFrMUJaekRIa0RUdkN0QXdFdkhPQzk0THE1blRwenNSKzk2TW8zZ3dpb0xqekR3ZTNXClJ2b21HMHVmQWhDMG5JOE04cWFQendHVDZIUEtkNDdDYjhFKzNlRnl1Y09KRC9OZncxMlBGU0tibmh6bHMxMkMKQ2pST253NzB4a1ZLUHRIeHBURDl3LzZMakJTYjlURjlISzZRVU1TQ3UzNWtOZCtjTk9KbFhncmdBUUtCZ0hVNApra003MGZRTmsrck1TWVd5eXBHeTBOeDUvUGF6dGhRU3pEeHhoVzc4OXlQZU5xcnJDQjlRYW9oUkJ1bUdyV2xHCkpxK1Z3SGpaM2twVmVMMGlCc0JLcU00MUp6cHJhaC80TUJWaVpWMnJvMGFNVXlnbW84SXNFclltb2JiUkxac2gKditjWXBQZE9nWU9KNGhCNFVTbmVlYXkrdllCQitmTTViR09nOTlJbEFvR0JBSm9jV2xGV0Z0TVdBa2t5VFF2SQplUklacEdibmQ1TDA1Z01LQ2pjOVVBVWFZTHBiMVRPOFFxejJWekc3V3VwclBabktqcitkNTlOSEwzQndIU1NECmVhVGdpZmVDNVlJTUVqYWFOOEJ6MDJsYnc2Qk1TOW42bnNkMUQ5Y2crRVd6bFFTOEVUV2lpR20xM284OXpaTUsKeENnUVNRR040QXdpVE9vU3ZFMVVWTFAxCi0tLS0tRU5EIFBSSVZBVEUgS0VZLS0tLS0=
```

Comment:
Ideally these listener cert/key pairs should be in individual `Secret` resources. shall we use use cert manager `Certificate` resources to define them? We have followed the same in all other places in the helm chart.

Reply:
yes, it will be changed when adding reading from cert ref implementation

Reply:
issue to track: https://github.com/wso2/apk/issues/949